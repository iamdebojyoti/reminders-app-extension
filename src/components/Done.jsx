import PropTypes from "prop-types";
import { Accordion, Card, Button } from "react-bootstrap";
import { Trash } from "./Icons";
import AccordionToggle from "./AccordionToggle";

const Done = ({reminders, remove}) => {
  return (
    <>
      {reminders.map((reminder, idx) => {
        return (
          <Accordion defaultActiveKey="0">
            <Card
              bg={"success"}
              key={idx}
              text={"light"}
              className="mb-2"
            >
              <Card.Header>
                {reminder.name}
                <Button
                  id={"remove-"+reminder.id}
                  className="float-end"
                  variant="success"
                  onClick={() => remove(reminder.id)}
                >
                  <Trash />
                </Button>
                <AccordionToggle eventKey={idx} />
              </Card.Header>
              <Accordion.Collapse eventKey={idx}>
                <Card.Body>
                  <Card.Text>{reminder.description}</Card.Text>
                </Card.Body>
              </Accordion.Collapse>
            </Card>
          </Accordion>
        );
      })}
    </>
  );
};

Done.propTypes = {
  remove: PropTypes.func.isRequired,
  reminders: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      state: PropTypes.string.isRequired,
    })
  ),
};

export default Done;
