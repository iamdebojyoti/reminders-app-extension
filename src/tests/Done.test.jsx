import { shallow } from "enzyme";
import Done from "../components/Done";

describe('Testing <Done/> component', () => {

    let component, props;

    beforeEach(() => {
        props = {
            remove: jest.fn(),
            reminders: [{
                id: '1',
                name: 'Reminder-1',
                description: 'Reminder-desc-1',
                state: 'created'
            }, {
                id: '2',
                name: 'Reminder-2',
                description: 'Reminder-desc-1',
                state: 'created'
            }],
        };
    });

    it('should match snapshot of Done at initial state', () => {
        component = shallow(<Done {...props}/>);
        expect(component.debug()).toMatchSnapshot();
    });

    it('should call done when trash button is clicked', () => {
        component = shallow(<Done {...props}/>);
        component.find('#remove-2').simulate('click');
        component.update();
        expect(props.remove).toHaveBeenCalledWith("2");
    })
});