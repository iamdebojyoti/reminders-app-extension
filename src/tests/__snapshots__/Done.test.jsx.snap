// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Done/> component should match snapshot of Done at initial state 1`] = `
"<Fragment>
  <Accordion defaultActiveKey=\\"0\\">
    <Card bg=\\"success\\" text=\\"light\\" className=\\"mb-2\\" body={false}>
      <CardHeader>
        Reminder-1
        <Button id=\\"remove-1\\" className=\\"float-end\\" variant=\\"success\\" onClick={[Function: onClick]} active={false} disabled={false}>
          <Trash />
        </Button>
        <AccordionToggle eventKey={0} variant=\\"success\\" />
      </CardHeader>
      <AccordionCollapse eventKey={0}>
        <CardBody>
          <CardText>
            Reminder-desc-1
          </CardText>
        </CardBody>
      </AccordionCollapse>
    </Card>
  </Accordion>
  <Accordion defaultActiveKey=\\"0\\">
    <Card bg=\\"success\\" text=\\"light\\" className=\\"mb-2\\" body={false}>
      <CardHeader>
        Reminder-2
        <Button id=\\"remove-2\\" className=\\"float-end\\" variant=\\"success\\" onClick={[Function: onClick]} active={false} disabled={false}>
          <Trash />
        </Button>
        <AccordionToggle eventKey={1} variant=\\"success\\" />
      </CardHeader>
      <AccordionCollapse eventKey={1}>
        <CardBody>
          <CardText>
            Reminder-desc-1
          </CardText>
        </CardBody>
      </AccordionCollapse>
    </Card>
  </Accordion>
</Fragment>"
`;
